compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -392,7 +432,11 @@ object Objects:
         // TODO: handle RefSet
         thisV match
         case ref: OfClass =>
-          resolveEnv(meth, ref.outer, ref.env)
+          ref.outer match
```

Comment:
@liufengyun I'm unsure about the possible cases for `ref.outer` here. `ref.outer` could be `Bottom` for top-level classes, could it ever be `Fun` or `RefSet`?

Reply:
As discussed in the email, it cannot be Fun, RefSet, OfArray.

Reply:
Then I think the current code behaves the same as the original implementation. The only case where `outer` is not a `ThisValue` is for top-level classes.